src/ggsn_gn.erl

```diff
@@ -673,7 +683,7 @@ copy_session_opts(K, Value, Opts)
 copy_session_opts(_K, _V, Opts) ->
     Opts.
 
-apply_session_opts(Opts, Session) ->
+apply_apn_defaults(Opts, Session) ->
```

Comment:
this can be further simplified, the options come directly from the IP pool config and are validated and normalized there. Converting the MS- attributes to binaries can happen there. The merge logic could be replace with a simple `maps:with/2`.

Reply:
Just to check that my understanding is correct. You would like to move the MS -attributes to binary conversion for the IPv4 addr to ergw_config (or ergw_gsn_lib?), then I can use the maps:with/2 in ggsn_sn, pgws5s8 an saegw_s11 to get the MS and Idle-timeout from APNOpts defaults. After that merge into Session (with Idle-timeout if received from ergw_aaa, session taking precedence).

Reply:
I had to touch that place in by Dual-Address-Bearer branch as well.
Some places where I made the changes:
* https://github.com/travelping/ergw/commit/2581ee4422b8999b370405cf932ea0e93479bdf8#diff-6008fcac06ab6921774337c87df68f8cR126
* https://github.com/travelping/ergw/commit/2581ee4422b8999b370405cf932ea0e93479bdf8#diff-b6a72d6a097437c19473134bfe2a1689R348
* https://github.com/travelping/ergw/commit/2581ee4422b8999b370405cf932ea0e93479bdf8#diff-b6a72d6a097437c19473134bfe2a1689L662